src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -257,8 +257,8 @@ public void putExternalResources(Set<String> locations) {
         final Set<ExternalResource> resources = loadExternalResources(locations);
         if (areExternalResourcesChanged(resources)) {
             reset();
+            fillCacheWithExternalResources(resources);
         }
-        fillCacheWithExternalResources(resources);
```

Comment:
The method call is important but it only really matters when `areExternalResourcesChanged` is `true`.

`areExternalResourcesChanged` checks if all resources changed or not present in cache. If all resources are there and not changed, then filling cache again with them is pointless. It is only after we call `reset` that we must re-populate the cache with external resources.

Reply:
confirmed